rigs/heft-web-rig/profiles/library/config/heft.json

```diff
@@ -4,55 +4,153 @@
+    //        * or "fileExtensions" setting.
+    //        */
+    //       "excludeGlobs": [],
+    //
+    //       /**
+    //        * Normally, when files are selected under a child folder, a corresponding folder will be created in
+    //        * the destination folder.  Specify flatten=true to discard the source path and copy all matching files
+    //        * to the same folder.  If two files have the same name an error will be reported.
+    //        * The default value is false.
+    //        */
+    //       "flatten": false,
+    //
+    //       /**
+    //        * If true, filesystem hard links will be created instead of copying the file.  Depending on the
+    //        * operating system, this may be faster. (But note that it may cause unexpected behavior if a tool
+    //        * modifies the link.)  The default value is false.
+    //        */
+    //       "hardlink": false
+    //     }
+    //   ]
+    // }
+
     {
-      /**
-       * The kind of built-in operation that should be performed.
-       * The "deleteGlobs" action deletes files or folders that match the
-       * specified glob patterns.
-       */
       "actionKind": "deleteGlobs",
-
-      /**
-       * The stage of the Heft run during which this action should occur. Note that actions specified in heft.json
-       * occur at the end of the stage of the Heft run.
-       */
       "heftEvent": "clean",
-
-      /**
-       * A user-defined tag whose purpose is to allow configs to replace/delete handlers that were added by other
-       * configs.
-       */
       "actionId": "defaultClean",
-
-      /**
-       * Glob patterns to be deleted. The paths are resolved relative to the project folder.
-       * Recommend exactly matching with "projectOutputFolderNames" in rush-project.json.
-       */
-      "globsToDelete": ["dist", "lib", "lib-commonjs", "temp"]
+      "globsToDelete": ["dist", "lib", "lib-amd", "lib-commonjs", "lib-es6", "temp"]
     }
   ],
 
   /**
    * The list of Heft plugins to be loaded.
    */
   "heftPlugins": [
-    {
-      /**
-       * The path to the plugin package.
-       */
-      "plugin": "@rushstack/heft-webpack4-plugin"
-
-      /**
-       * An optional object that provides additional settings that may be defined by the plugin.
-       */
-      // "options": { }
-    },
-    {
-      "plugin": "@rushstack/heft-jest-plugin"
-    },
-    {
-      "plugin": "@rushstack/heft-sass-plugin"
-    }
+    // {
+    //  /**
+    //   * The path to the plugin package.
+    //   */
+    //  "plugin": "path/to/my-plugin",
+    //
+    //  /**
+    //   * An optional object that provides additional settings that may be defined by the plugin.
+    //   */
+    //  // "options": { }
+    // }
+    { "plugin": "@rushstack/heft-jest-plugin" },
+    { "plugin": "@rushstack/heft-sass-plugin" }
+    // { "plugin": "@rushstack/heft-webpack5-plugin" }
```

Comment:
Yes, isn't the common practice today?

Reply:
✔ We agreed to keep this as-is until Rush Stack has a recommended solution for bundling library projects.

Reply:
🤔 However I do think we should add a `heft-web-rig/profiles/library/webpack-base.config.js` for library projects that want to opt-in to bundling.

Reply:
@iclanton in our chat, we observed that today most Rush libraries are NOT bundled, because Rush Stack does not have a formal model yet for how to prevent the app from rebundling its libraries.  So the plan was to keep this as-is.

However I realized that:
1. If a library project does want to opt-in to bundling, it cannot add `{ "plugin": "@rushstack/heft-webpack5-plugin" }` in its own `config/heft.json` because Heft would try to resolve the `heft-webpack5-plugin` package from the local project folder (not the rig), AND
2. Enabling this line in the rig is fine, because the Webpack plugin will be skipped unless the library project has a **webpack.config.js** file (which is the mechanism for opting in)
3. We need to add a `heft-web-rig/profiles/library/webpack-base.config.js` file to support this scenario.

I'm going to do that.  I'm also going to add a test project for each of the `heft-web-rig` profiles.

Reply:
There's nothing wrong with the app rebundling its libraries; that produces more optimized outputs. Doesn't mean we can't still bundle the library to optimize unit tests (only loading 1 file will usually be better for performance than many) or similar.

Reply:
Also, doesn't this enable the library to specify a default bundling for itself, which will take effect if the app does not opt to re-bundle it?

Reply:
> There's nothing wrong with the app rebundling its libraries; that produces more optimized outputs. 

The main problem with that is build times.  We want the computational cost for building a Rush project to be `O('lib/*.js')`, not `O('node_modules/*.js')`.

BTW if the app wants to rebundle, it should probably use the `lib/index.js` ESM entry point instead of the `dist/bundle.js` default entry point from **package.json**.

Reply:
> Also, doesn't this enable the library to specify a default bundling for itself, which will take effect if the app does not opt to re-bundle it?

Yes 👍